libtest/src/main/java/tests/custom/set/SynchronizedMethodSet.kt

```diff
@@ -0,0 +1,37 @@
+/*-
+ * #%L
+ * Lincheck
+ * %%
+ * Copyright (C) 2019 JetBrains s.r.o.
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+package tests.custom.set
+
+class SynchronizedMethodSet {
```

Comment:
Why do you need several sets with global locks?

Reply:
To test the support of all ways of locking

Reply:
Probably, it would be better to have a sequential implementation and inherit all these lock-based implementations from it. Please, also put all of them in a single file.